src/Appium.Net/Appium/Android/UiAutomator/AndroidUiScrollable.cs

```diff
@@ -41,14 +42,316 @@ public AndroidUiScrollable()
+        {
+            if (maxSwipes < 0)
+                throw new ArgumentOutOfRangeException(nameof(maxSwipes), "Must be 0 (zero) or greater");
+
+            var forkedBuilder = new StringBuilder(_builder.ToString());
+            forkedBuilder.AppendFormat(".flingToBeginning({0})", maxSwipes);
+            return new TerminatedStatementBuilder(forkedBuilder);
+        }
+
+        /// <summary>
+        /// Performs a fling gesture to reach the end of a scrollable layout element. The end can be at the
+        /// bottom-most edge in the case of vertical controls, or the right-most edge for horizontal controls.
+        /// Make sure to take into account devices configured with right-to-left languages like Arabic and Hebrew.
+        /// Maps to the UiScrollable.flingToEnd() method.
+        /// </summary>
+        /// <param name="maxSwipes"></param>
+        /// <remarks>https://developer.android.com/reference/android/support/test/uiautomator/UiScrollable#flingtoend</remarks>
+        public TerminatedStatementBuilder FlingToEnd(int maxSwipes)
+        {
+            if (maxSwipes < 0)
+                throw new ArgumentOutOfRangeException(nameof(maxSwipes), "Must be 0 (zero) or greater");
+
+            var forkedBuilder = new StringBuilder(_builder.ToString());
+            forkedBuilder.AppendFormat(".flingToEnd({0})", maxSwipes);
+            return new TerminatedStatementBuilder(forkedBuilder);
+        }
+
+        /// <summary>
+        /// Searches for a child element in the present scrollable container. The search first looks for a
+        /// child element that matches the selector you provided, then looks for the content-description in
+        /// its children elements. If both search conditions are fulfilled, the method returns a {@ link UiObject}
+        /// representing the element matching the selector (not the child element in its subhierarchy containing
+        /// the content-description). By default, this method performs a scroll search.
+        /// Maps to the UiScrollable.getChildByDescription(UiSelector, String, boolean) method.
+        /// </summary>
+        /// <param name="uiSelector">
+        /// UiSelector for a child in a scollable layout element
+        /// </param>
+        /// <param name="description">
+        /// Content-description to find in the children of the childPattern match (may be a partial match)
+        /// </param>
+        /// <param name="allowScrollSearch">
+        /// Set to true if scrolling is allowed
+        /// </param>
+        /// <remarks>https://developer.android.com/reference/android/support/test/uiautomator/UiScrollable#getchildbydescription_1</remarks>
+        public TerminatedStatementBuilder GetChildByDescription(AndroidUiSelector uiSelector, string description, bool allowScrollSearch = true)
+        {
+            if (uiSelector == null)
+                throw new ArgumentNullException(nameof(uiSelector));
```

Comment:
could the https://stackoverflow.com/questions/29184887/best-way-to-check-for-null-parameters-guard-clauses syntax be used instead to make the code shorter?

Reply:
In general, I like the Kotlin's approach to it: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/require.html

Reply:
A project I work on uses the extension method approach for their null guards, has a similar effect.

Example of the way I was going to handle the non-neg guard from you other comment:
```csharp
public void aaaa(int value)
{
    value.EnsureNumberIsPositive(nameof(value));
    // etc etc etc...
}
```

If you like this approach, I can do the null guards the same way. The one downside is that the top of the stack trace will have the guard method rather than the actual method the user called. Though it's pretty minor since the stack will at least still have the user-called method on it.

Reply:
I simply see an unnecessary code duplication, which could be easily eliminated.  
It would be also effective to return the same value from the verification method if the verification passes. This way we could also save one more line of code